### DISCUSSION TITLE
Release build crashes caused by XAML issues

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dotnet/maui#18532] [General]
Hi all

We are having several issues where everything works ok on our local devices, but then crashes when we deploy release builds. 

We have identified this is mostly caused by bugs in our XAML code.

I am guessing there is some difference with XAML in release vs debug build (possibly compilation?). The issue is debug seems to be more forgiving with problems, which means it is only after release that they come to light.

The problem is these bugs are very hard to find. The stack trace shows you nothing of value, so you need to keep doing repeated releases with additional error finding code which is immensely time consuming.

Does anyone know a) what is causing this, and b) what I can do about it to find these issues while debugging?

I have set this in App:
[assembly: XamlCompilation(XamlCompilationOptions.Compile)]

I have also added this to the code behind (not sure I need to because of the above?):
[XamlCompilation(XamlCompilationOptions.Compile)]

Should I be doing something else?

Thanks

**Comment by NeilN1** [comment]
How are you getting the stack traces? We've implemented Raygun's MAUI crash reporting and it reports back what XAML caused the crash (for some crashes) in release builds.

**Reply by williambuchanan2** [reply]
We are using App Centre diagnostics, but I also put an email sender in the exception handler which sends me an email with the stack trace. The information we get is pretty useless though. For example the last problem we had was caused by a binding error, but the exception being thrown was  'unable to figure out route for: //MyPage'. It seems to throw an exception on the symptom rather than the cause.
